### PR TITLE
2.1.0 Beta

### PR DESCRIPTION
**Pull request by silverman63**
This establishes the 2.1.0 beta, providing a new mass action that can theoretically send all products for a given store to the Sailthru Content Library at once. May take multiple minutes, and may run into memory issues. Please report feedback if so.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,16 +27,26 @@ Documentation is available for the extension and our APIs on the [Sailthru GetSt
...
+* The 2.0.0 uses updated schemas for User and Template Vars. Visit the documentation above to see the full schemas and what's changed.
 
 ## Installation 
-* *We've moved codepools. Before installing, make sure to remove the current plugin.*
 * *After installation, you may need to login, clear cache, and then re-login.*
 
 ### Manual Installation
 
-#### With Modman
+#### (Migration-only) Removing the old extension
```
Comment (nickgundry) [inline README.md]: Is removing the old extension referring to 1.x or 2.x branches in this instance?

File: app/code/community/Sailthru/Email/etc/config.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 <config>
     <modules>
         <Sailthru_Email>
-            <version>2.1.0</version>
+            <version>2.1.0-beta</version>
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/etc/config.xml]: Would this be the release version displayed? And will you update this to remove beta when merged into master?

File: app/code/community/Sailthru/Email/controllers/ContentController.php [hunk 3]
```diff
@@ -1,8 +1,10 @@
 <?php
 class Sailthru_Email_ContentController extends Mage_Adminhtml_Controller_Action
 {
-
-    const MAX_ITEM_POST = 350;
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/controllers/ContentController.php]: Why the removal of the upper limit? Will this handle thousands of products now?

File: README.md [hunk 4]
```diff
@@ -14,8 +14,8 @@ The Magento extension provides an integration to the Sailthru platform and can s
 
 ##### Beta Features
 - **Send to Sailthru Mass Action**
-Allows for bulk product update being sent to the Sailthru Content Library from the admin product catalog.
-*May run into memory issues.*
+    A new mass action that sends bulk product updates to Sailthru from the admin product catalog screen.
+    *May run into memory issues.*
```
Comment (nickgundry) [inline README.md]: +1

File: var/connect/Sailthru_Email.xml [hunk 5]
```diff
@@ -18,9 +18,11 @@
 </description>
    <license>Apache License 2.0</license>
    <license_uri>http://www.apache.org/licenses/LICENSE-2.0</license_uri>
-   <version>2.0.0</version>
-   <stability>stable</stability>
-   <notes>Large update adding many new features, more granular configuration, auto-tagging, and resolving issues with API connectivity, For more information, visit our documentation at https://getstarted.sailthru.com/integrations/magento/magento-1-extension-v2/.</notes>
+   <version>2.1.0-beta</version>
```
Comment (nickgundry) [inline var/connect/Sailthru_Email.xml]: Same as previous question about labels. Will we update this when moving to master and prior to releasing.

**Review by nickgundry — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (silverman63) [reply README.md]: Removing 1.x (hence migration - goes with above section on migration from 1.x)

File: var/connect/Sailthru_Email.xml (hunk 5 above)
Comment (silverman63) [reply var/connect/Sailthru_Email.xml]: Yup

File: README.md [hunk 6]
```diff
@@ -37,7 +37,7 @@ For questions or troubleshooting, please visit [Sailthru Support][5] or submit a
 
 ### Manual Installation
 
-#### (Migration-only) Removing the old extension
+#### (Migration from 1.x only) Removing the old extension
```
Comment (nickgundry) [inline README.md]: +1

File: README.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
 # Sailthru Magento Extension
-### Version 2.0.0
+### Version 2.1.0 - Beta
```
Comment (nickgundry) [inline README.md]: +1

File: README.md [hunk 8]
```diff
@@ -51,21 +66,19 @@ modman clone git://github.com/sailthru/sailthru-magento-extension.git
 modman update sailthru-magento-extension # to update
 ```
 
-#### Without Modman
+#### Installation without modman
```
Comment (nickgundry) [inline README.md]: +1

File: app/code/community/Sailthru/Email/controllers/ContentController.php [hunk 9]
```diff
@@ -23,35 +29,46 @@ public function bulkAction() {
...
+                        );
+                }
             }
-            usleep(0.25 * 1000000);
-        }
+
+            $collection->clear();
+        } while ($page <= $collection->getLastPageNumber());
+
         $endTime = microtime(true);
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/controllers/ContentController.php]: +1

File: app/code/community/Sailthru/Email/controllers/ContentController.php [hunk 10]
```diff
@@ -23,35 +29,46 @@ public function bulkAction() {
...
+                    if ($success) {
+                        $savedProducts++;
+                    }
+                } catch (Sailthru_Client_Exception $e) {
+                    Mage::getSingleton('adminhtml/session')
+                        ->addError(
+                            "Error saving {$product->getName()} ({$product->getSku()})" .
+                            "<pre >({$e->getCode()}) {$e->getMessage()}</pre>"
+                        );
+                }
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/controllers/ContentController.php]: +1

File: app/code/community/Sailthru/Email/etc/config.xml [hunk 11]
```diff
@@ -9,7 +9,7 @@
 <config>
     <modules>
         <Sailthru_Email>
-            <version>2.0.0</version>
+            <version>2.1.0-beta</version>
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/etc/config.xml]: +1

File: var/connect/Sailthru_Email.xml [hunk 12]
```diff
@@ -18,9 +18,11 @@
...
+   <stability>beta</stability>
+   <notes>This update enables a Beta feature for adding a new Sailthru mass action for the product catalog. Send bulk updates to the Sailthru content library for a given store. 
+
+* Note: This is beta functionality and may take longer than two minutes to complete, you may also see memory issues on the request. Please report any issues. *</notes>
```
Comment (nickgundry) [inline var/connect/Sailthru_Email.xml]: +1

File: var/connect/package.xml [hunk 13]
```diff
@@ -17,11 +17,13 @@
 - Capture customers order data with adjustments and Sailthru campaign attribution.&#xD;
 - Deliver abandoned cart emails for known users.&#xD;
 </description>
-    <notes>Large update adding many new features, more granular configuration, auto-tagging, and resolving issues with API connectivity, For more information, visit our documentation at https://getstarted.sailthru.com/integrations/magento/magento-1-extension-v2/.</notes>
+    <notes>This update enables a Beta feature for adding a new Sailthru mass action for the product catalog. Send bulk updates to the Sailthru content library for a given store. &#xD;
```
Comment (nickgundry) [inline var/connect/package.xml]: +1

File: var/connect/package.xml [hunk 14]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0"?>
 <package>
     <name>Sailthru_Email</name>
-    <version>2.0.0</version>
-    <stability>stable</stability>
+    <version>2.1.0-beta</version>
+    <stability>beta</stability>
```
Comment (nickgundry) [inline var/connect/package.xml]: +1

File: var/connect/package.xml [hunk 15]
```diff
@@ -17,11 +17,13 @@
...
...="2d4ade7f625c0a192b6ddbf11f734746"/></dir><dir name="etc"><file name="adminhtml.xml" hash="9d58ae88d6fd3784099f6a08b46920bf"/><file name="config.xml" hash="6395931902556d35920a1fbef5e064c6"/><file name="system.xml" hash="174ef568fa050f02c3579e939710d6ba"/></dir><dir name="lib"><file name="Sailthru_Client.php" hash="7304b694e3472f61512b7a34a9f8604c"/><file name="Sailthru_Client_Exception.php" hash="465d0a188a17061147003e5e8aa22331"/><file name="Sailthru_Util.php" hash="0bdae6568a14bc981905cb1d428bce26"/></dir><dir name="sql"><dir name="sailthru_email_setup"><file name="install-2.0.0.php" hash="65466689029b9b9dc0347999ed080a8b"/></dir></dir></dir></dir></target><target name="mageetc"><dir name="modules"><file name="Sailthru_Email.xml" hash="243c2046f5076c1f2c25bcea38ea6e06"/></dir></target><target name="magedesign"><dir name="frontend"><dir name="base"><dir name="default"><dir name="layout"><file name="sailthru.xml" hash="fa0a38b14361712ed4e790bbc06fcacd"/></dir><dir name="template"><dir name="sailthru"><file name="Horizon.phtml" hash="626724055cd5a9a9799c9959fe28b06d"/><file name="Sst.phtml" hash="9263c251000893c9c502a8d6ae9e1542"/></dir></dir></dir></dir></dir></target></contents>
```
Comment (nickgundry) [inline var/connect/package.xml]: +1

**Review by nickgundry — APPROVED** [review]

**Review by nickgundry — APPROVED** [review]

**Pull request merged**